velox/connectors/hive/storage_adapters/s3fs/S3FileSystem.cpp

```diff
@@ -170,6 +171,18 @@ class S3Config {
     return {};
   }
 
+  std::optional<std::string> iamRole() const {
+    if (config_->isValueExists("hive.s3.iam-role")) {
+      return config_->get("hive.s3.iam-role").value();
+    }
+    return {};
+  }
+
+  std::string iamRoleSessionName() const {
+    return config_->get(
+        "hive.s3.iam-role-session-name", std::string("velox-session"));
```

Comment:
nit: I wonder if these string constants should be pulled into their own class say S3Config's so that  these config strings can be centralized and documentation on them added .

Reply:
@kgpai I have an issue open for documentation: https://github.com/facebookincubator/velox/issues/864
I will make the necessary changes required there. Do we currently have any examples of such parameters?